onnx/defs/nn/defs.cc

```diff
@@ -164,25 +164,19 @@ void convPoolShapeInference(
       continue;
     }
     // how big is the input, including padding
-    int64_t effective_input_size = input_shape.dim(2 + i).dim_value();
-    effective_input_size += pads[i];
-    effective_input_size += pads[i + kernel_shape_size];
+    int64_t input_size = input_shape.dim(2 + i).dim_value();
+    int64_t effective_input_size = input_size + pads[i] + pads[i + kernel_shape_size];
 
     // default is floor mode .i.e. ceil_mode is set to 0
     auto ceil_mode = getAttribute(ctx, "ceil_mode", 0);
 
-    // how many times we can move the kernel from it's initial position, based
-    // on the stride
-    int64_t strided_kernel_positions;
-
-    if (ceil_mode == 1)
-      strided_kernel_positions =
-          (int64_t)(std::ceil((effective_input_size - effective_kernel_shape[i]) / float(strides[i])));
-    else
-      strided_kernel_positions = (effective_input_size - effective_kernel_shape[i]) / strides[i];
+    int64_t output_size =
+        (effective_input_size - effective_kernel_shape[i] + (ceil_mode ? strides[i] - 1 : 0)) / strides[i] + 1;
```

Comment:
Should this be documented?

Reply:
Agree. Well, "documentation" could mean different things here. As a first step, it would be great to explain the computation clearly here in the code (which can be used in the operator documentation with suitable modification).

Reply:
I assume the goal is to have the same behavior as in the torch op (which is linked to in the ONNX documentation). I guess the relevant line from the torch documentation which may be worth adding to the ONNX op documentation is this
```
When ceil_mode=True, sliding windows are allowed to go off-bounds if they start within the
left padding or the input. Sliding windows that would start in the right padded region are ignored.
```